lib/profileAgents.js

```diff
@@ -0,0 +1,356 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const assert = require('assert-plus');
+const base64url = require('base64url-universal');
+const bedrock = require('bedrock');
+const crypto = require('crypto');
+const {CapabilityAgent} = require('webkms-client');
+const database = require('bedrock-mongodb');
+const {promisify} = require('util');
+const kms = require('./kms');
+const utils = require('./utils');
+const zcaps = require('./zcaps');
+
+const {util: {BedrockError}} = bedrock;
+
+// load config defaults
+require('./config');
+const KMS_MODULE = 'ssm-v1';
+
+bedrock.events.on('bedrock-mongodb.ready', async () => {
+  await promisify(database.openCollections)(['profile-profileAgent']);
+
+  await promisify(database.createIndexes)([{
+    // cover queries of a profile agent by its ID
+    collection: 'profile-profileAgent',
+    fields: {'profileAgent.id': 1},
+    options: {unique: true, background: false}
+  }, {
+    // cover queries of a profile agent by its ID
```

Comment:
comment needs to be updated or removed.  I favor removal for this exact reason, comments are often just more code to maintain.